crates/bevy_winit/src/lib.rs

```diff
@@ -360,45 +361,25 @@ pub fn winit_runner(mut app: App) {
                 }
             }
         }
-        runner_state.redraw_requested = false;
 
         match event {
-            Event::NewEvents(start_cause) => match start_cause {
-                StartCause::Init => {
-                    #[cfg(any(target_os = "ios", target_os = "macos"))]
-                    {
-                        let (
-                            commands,
-                            mut windows,
-                            event_writer,
-                            winit_windows,
-                            adapters,
-                            handlers,
-                            accessibility_requested,
-                        ) = create_window_system_state.get_mut(&mut app.world);
-
-                        create_windows(
-                            event_loop,
-                            commands,
-                            windows.iter_mut(),
-                            event_writer,
-                            winit_windows,
-                            adapters,
-                            handlers,
-                            accessibility_requested,
-                        );
-
-                        create_window_system_state.apply(&mut app.world);
+            Event::AboutToWait => {
```

Comment:
Whats the reason for moving `Window::request_redraw()` into `Event::AboutToWait`?

I would recommend to call `Window::request_redraw()` as soon as you know that you want to draw. Calling it in `Event::AboutToWait` has the disadvantage of queuing that request later then you could, which might let you miss frames.

The previous method of calling `Window::request_redraw()` at the end of the event loop was therefor better imo.

Reply:
the previous method doesn't work on iOS, because most of Bevy request redraw come from the update which is called when reacting to a `RedrawRequested` event, and that doesn't work on iOS

Reply:
What exactly doesn't work on iOS?
Are you saying that `WindowEvent::RedrawRequested` is not being sent on iOS?

Reply:
Yup, `WindowEvent::RedrawRequested` is not sent if `request_redraw()` is called while processing `WindowEvent::RedrawRequested`.

I also saw this assert in winit code that also makes me believe it should be avoided
https://github.com/rust-windowing/winit/blob/master/src/platform_impl/ios/app_state.rs#L648-L651

Reply:
Yeah, this would be a bug in Winit.
Will report back when I know more!

Reply:
@mockersf would you mind testing https://github.com/daxpedda/winit/tree/ios-fix-request-redraw-from-redraw-requested-v0.29 and checking if this solves the problem?

Reply:
@daxpedda I haven't had time yet to test your branch and Bevy is currently broken for another reason on iOS. I will as soon as possible and update you!